docs/durable_execution/temporal.md

```diff
@@ -182,6 +188,112 @@ As the streaming model request activity, workflow, and workflow execution call a
 - To get data from the workflow call site or workflow to the event stream handler, you can use a [dependencies object](#agent-run-context-and-dependencies).
 - To get data from the event stream handler to the workflow, workflow call site, or a frontend, you need to use an external system that the event stream handler can write to and the event consumer can read from, like a message queue. You can use the dependency object to make sure the same connection string or other unique ID is available in all the places that need it.
 
+### Model Selection at Runtime
+
+[`Agent.run(model=...)`][pydantic_ai.agent.Agent.run] normally supports both model strings (like `'openai:gpt-5.2'`) and model instances. However, `TemporalAgent` does not support arbitrary model instances because they cannot be serialized for Temporal's replay mechanism.
+
+Instead, you can:
+
+1. **Use model strings**: Pass a model string like `'openai:gpt-5.2'` to `temporal_agent.run(model=...)`, which will be instantiated at runtime.
+2. **Pre-register model instances**: Pass a dict of model instances to `TemporalAgent(models={...})`, then reference them by name (e.g., `temporal_agent.run(model='fast')`) or by passing the same instance (e.g., `temporal_agent.run(model=fast_model)`).
```

Comment:
If we're going to explicitly mention `'fast'` and `fast_model`, I think we should have them in the first `models={}` snippet as well.

But I'm also OK with not having inline examples here at all, just saying "by name or ... instance"